lib/matplotlib/axes/_axes.py

```diff
@@ -7369,12 +7396,44 @@ def _kde_method(X, coords):
             kde = mlab.GaussianKDE(X, bw_method)
             return kde.evaluate(coords)
 
-        vpstats = cbook.violin_stats(dataset, _kde_method, points=points)
-        return self.violin(vpstats, positions=positions, vert=vert,
+        def _resolve_duplicate_alias(func_obj_list):
+            unique_alias_set = set()
+            for func_obj in func_obj_list:
+                while func_obj.alias in unique_alias_set:
+                    func_obj.alias += 'x'
+                unique_alias_set.add(func_obj.alias)
+
+        violin_stat_func_obj_list = []
+        for func in statistics_function_list:
+            if not isinstance(func, ViolinStatFunc):
+                if callable(func):
+                    violin_stat_func_obj_list.append(ViolinStatFunc(func))
+                else:
+                    raise ValueError(
+                        'Optional argument has to be a callable' +
+                        'or a ViolinStatFunc object')
+            else:
+                violin_stat_func_obj_list.append(func)
+
+        custom_stat_alias_list = [func_obj.alias for func_obj in
+                                  violin_stat_func_obj_list]
+        if len(custom_stat_alias_list) > len(set(custom_stat_alias_list)):
+            _resolve_duplicate_alias(violin_stat_func_obj_list)
+            # remake alias list based on updated unique aliases
+            custom_stat_alias_list = [func_obj.alias for func_obj in
+                                      violin_stat_func_obj_list]
+
+        vpstats, custom_stat_vals = \
+            cbook.violin_stats(dataset, _kde_method,
+                               violin_stat_func_obj_list, points=points)
+
+        return self.violin(vpstats, custom_stat_vals, custom_stat_alias_list,
+                           positions=positions, vert=vert,
                            widths=widths, showmeans=showmeans,
                            showextrema=showextrema, showmedians=showmedians)
 
-    def violin(self, vpstats, positions=None, vert=True, widths=0.5,
+    def violin(self, vpstats, custom_stat_vals, custom_stat_alias_list,
```

Comment:
Are the custom stat-related values required parameters now? They should be be optional, IMO